maven-simplest/src/main/java/io/vertx/example/HelloServer.java

```diff
@@ -0,0 +1,62 @@
+package io.vertx.example;
+
+import java.time.Instant; 
+import io.vertx.core.Vertx;
+import io.vertx.ext.web.Router;
+
+
+public class HelloServer {
+
+    private int counter;
+    private float maxRequestsPerSecond;
+    private String greeting;
+    private Instant lastRequestInstant;
+    private Vertx vertx;
+
+    HelloServer(String greeting, float maxRequestsPerSecond) {
+        this.greeting = greeting;
+        this.maxRequestsPerSecond = maxRequestsPerSecond;
+        lastRequestInstant = Instant.MIN;
+        vertx = Vertx.vertx();
+    }
+
+    public void start() {
+        vertx
+            .createHttpServer()
+            .requestHandler(defineRoutes())
+            .listen(8080);
+    }
+
+    private Router defineRoutes() {
+        Router router = Router.router(vertx);
+
+        router.get("/").handler(req -> {
+            Instant now = Instant.now(); 
+            if (verifyRateLimit(now)) {
+                req.response().end(greeting + "\n");
+            } else {
+                req.response().setStatusCode(503).end();
+            }
+
+            lastRequestInstant = now;
+            counter++;
```

Comment:
The access to `counter` works as long as you don't share it among different threads. From what I know so far (not much, I'm just getting started with Vert.x), Vert.x places each _Verticle_ in its own thread. In this case, I think that we only have a single verticle, hence a single thread.

But what if we had multiple instances of the same verticle, would this work?

Reply:
I found a way to reproduce the scenario of having multiple instances of the same verticle, so that each of them updates its own counter:

```
        int eventLoopSize = 16;
        Vertx vertx = Vertx.vertx();
        DeploymentOptions options = new DeploymentOptions().setInstances(eventLoopSize);
        vertx.deployVerticle("io.vertx.example.HelloServer", options);
```

Reply:
I would not care about the number of verticles here. This is a toy project, so we are not using multiple verticles.